feature-libs/asm/components/customer-emulation/customer-emulation.component.ts

```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
     );
     this.isCustomerEmulationSessionInProgress$ =
       this.asmComponentService.isCustomerEmulationSessionInProgress();
+
+    this.cartId.valueChanges.subscribe((response: string) => {
+      this.cartIdExists = response.trim().length > 0;
+    });
+
+    this.activeCartFacade.getActiveCartId().subscribe((response) => {
+      if (response) {
+        this.cartId.setValue(response);
+      }
+    });
   }
 
   logoutCustomer() {
     this.asmComponentService.logoutCustomer();
   }
 
+  /**
+   * Assign the input cart number to the customer
+   */
+  assignCartToCustomer() {
+    const customerId = this.customer.uid;
+    const cartId = this.cartId.value;
+
+    if (customerId) {
+      this.occAsmAdapter.bindCart(cartId, customerId).subscribe(
+        () => {
+          this.showAssignCartSuccess = true;
+
+          setTimeout(() => {
+            this.showAssignCartSuccess = false;
+            this.submitEvent.emit({ customerId: customerId });
+          }, 3000);
+        },
+        () => {
+          this.showAssignCartError = true;
+
```

Comment:
Other behaviour question: there is a cart id pre-populated in the cart binding input field when it's displayed.  Where does that come from and what was the intent?  (also, it resulted in an error when I tried to bind it.)

Reply:
The cart id that is pre-populated is the same functionality as Accelerator. It displays the current assigned cart for the selected customer. It fails because the cart is already assigned to a customer.